src/util/MmapVectorImpl.h

```diff
@@ -77,14 +80,14 @@ void MmapVector<T>::readMetaDataFromEnd() {
 template <class T>
 void MmapVector<T>::mapForReading() {
   // open to get valid file descriptor
-  FILE* fp = fopen(_filename.c_str(), "r");
+  int orig_fd = ::open(_filename.c_str(), O_RDONLY);
```

Comment:
This is just to make things more similar to the write case

Reply:
totally agreed.